lib/core/storage/clientAdapter.js

```diff
@@ -462,6 +462,21 @@ class ClientAdapter {
         return this.client.exists(index, collection, id);
       });
 
+    /**
+     * Check if a document multiple document Exists
+     *
+     * @param {string} index
+     * @param {string} collection
+     * @param {string} id -- document unique identifier
+     * @returns {Promise.<boolean>}
+     */
+    global.kuzzle.onAsk(
+      `core:storage:${this.scope}:document:mExist`,
+      (index, collection, ids) => {
+        this.cache.assertCollectionExists(index, collection);
+        return this.client.mGet(index, collection, ids);
```

Comment:
I'm not sure that calling `mGet` is the best way to verify multiple documents exists, at least it shouldn't return document sources otherwise this is not better than calling `document:mGet`

Reply:
This is because mGet is a request to ES and there is no mExists request in ES. I will add filters on the mGet request in mExists to have a mExists action that gives only the existence of the documents and no extra info

Reply:
I agree with @Shiranuit , the Elasticsearch `mGet` API action provide a query parameter `_source` to enable/disable documents sources fetching. You should use it and set it to `false`